docs/repo/property-pages/property-specification.md

```diff
@@ -0,0 +1,220 @@
+﻿# Property Specification
+
+This document details how properties are specified, which controls their appearance and behavior in the Project Properties UI and how they are written to the project file.
+
+## XAML Rule Files
+
+The set of properties to display in the UI are outlined declaratively in XAML files that ship with the Project System. This means that most modifications to the Project Properties UI can be achieved by simply modifying an XML file.
+
+Here we will walk through the structure of these files. Some familiarity with XAML rule files is assumed. We will not discuss data sources here.
+
+Each XAML rule file describes a single page of properties. Multiple pages are displayed at once, so there are multiple rule files involved in the end-to-end experience. 
+
+Here is a complex example of a string property that demonstrates the majority of features we will discuss below.
+
+```xml
+<StringProperty Name="MyProperty"
+                DisplayName="My property"
+                Description="This is my property."
+                HelpUrl="https://helpful.com/all-about-my-property"
```

Comment:
Should this be a domain we control? Or example.com?

Reply:
```suggestion
                HelpUrl="https://example.com/all-about-my-property"
```